packages/_tests/test_packages_common.py

```diff
@@ -35,78 +23,69 @@ def registered_packages() -> list[str]:
+
+
+def generate_test_list(lockfile: PyodideLockSpec) -> list[ImportTestCase]:
+    packages = lockfile.packages
+    testcases: list[ImportTestCase] = [
+        {
+            "name": package.name,
+            "imports": [normalize_import_name(name) for name in package.imports],
+        }
+        for package in packages.values()
+        if package.package_type in ("package", "cpython_module")
+    ]
+
+    return testcases
+
+
+@functools.cache
+def build_testcases():
+    lockfile = load_lockfile()
+    return generate_test_list(lockfile)
+
+
+def idfn(testcase: ImportTestCase) -> str:
+    # help pytest to display the name of the test case
+    return testcase["name"]
 
 
 @pytest.mark.skip_refcount_check
 @pytest.mark.driver_timeout(120)
-@pytest.mark.parametrize("name", registered_packages())
-@pytest.mark.benchmark(
-    max_time=3,
-    min_rounds=1,
-    timer=time.perf_counter,
-)
-def test_import(
-    name: str, selenium_standalone: _BrowserBaseRunner, benchmark: Any
-) -> None:
-    if not package_is_built(name):
-        raise AssertionError(
-            "Implementation error. Test for an unbuilt package "
-            "should have been skipped in selenium_standalone fixture"
-        )
+@pytest.mark.parametrize("testcase", build_testcases(), ids=idfn)
+def test_import(selenium_standalone, testcase: ImportTestCase):
+    name = testcase["name"]
+    imports = testcase["imports"]
 
     if name in XFAIL_PACKAGES:
         pytest.xfail(XFAIL_PACKAGES[name])
 
-    meta = MetaConfig.from_yaml(PKG_DIR / name / "meta.yaml")
-
     if name in UNSUPPORTED_PACKAGES[selenium_standalone.browser]:
         pytest.xfail(
             f"{name} fails to load and is not supported on {selenium_standalone.browser}."
         )
 
-    selenium_standalone.run("import glob, os, site")
-
-    def _get_file_count(expr):
-        return selenium_standalone.run(
-            f"""
-            len(list(glob.glob(
-                site.getsitepackages()[0] + '{expr}',
-                recursive=True)
-            ))
-            """
-        )
-
-    import_names = meta.test.imports
-    if not import_names:
-        import_names = meta.package.top_level
-
-    if not import_names:
-        # Nothing to test
+    if not imports:
         return
 
-    def _import_pkg():
-        for import_name in import_names:
-            selenium_standalone.run_async(f"import {import_name}")
-
-    benchmark(_import_pkg)
-
-    # Make sure that after importing, either .py or .pyc are present but not
-    # both
-    pyc_count = sum(_get_file_count(f"/{key}/**/*.pyc") for key in import_names)
-    py_count = sum(_get_file_count(f"/{key}/**/*.py") for key in import_names)
-
-    assert not ((py_count > 0) and (pyc_count > 0))
-
-    # Make sure no exe files were loaded!
-    assert _get_file_count("/**/*.exe") == 0
+    for import_name in imports:
+        selenium_standalone.load_package(name)
+        selenium_standalone.run(f"import {import_name}")
```

Comment:
I wonder if we could consider switching away from selenium_standalone for this. Or say doing several packages with the same selenium instance before rebooting it. Presumably loading all packages in one instance leads to OOM. But we used to be using selenium_standalone to validate that there are no missing deps. Now I think we are primarily concerned with validating the deps elsewhere and we're primarily concerned that the shared objects have the correct abi.

Reply:
No action needed here of course.

Reply:
Sounds reasonable to me. Thanks for the suggestion!